### PR TITLE
Introduce new help view to display information on any view

### PR DESCRIPTION
**Pull request by bIgBV**
As a part of addressing #225 , I'm introducing a new floating pane which can show information relevant to the current in focus pane. This can include shortcuts, key bindings and glossary information when using symbols and emojis..

### CODE REVIEW

File: console/src/view/mod.rs
Comment (bIgBV) [inline console/src/view/mod.rs]: Just realized that this is unreachable. I guess I will have to create a new `ViewState` state to indicate that the help state is present. I think as long as I store what the previous state was, I can go back to it when `esc` (or any other key) is pressed.

**Comment by bIgBV** [comment]
Does anyone have thoughts on what the contents of the help popup should be in different views? For now I'm working on the mechanism to pump the information from the view to the popup. This is what it looks like currently:

image

**Comment by hawkw** [comment]
@bIgBV this is awesome, thanks for working on this!

As for what should be displayed in the help text:

 - I think we should probably display the controls for the current view, even though they're also displayed in the header...perhaps once the help text feature is merged, we should consider hiding the controls from the header by default, and just display what key is needed to open the help text?
- It could be useful to explain the meanings of different emojis/symbols in the current view?
- Eventually, the same popup system could maybe be used for displaying detailed descriptions of warnings, but that's something we would add later (since we don't have a system for associating help with warnings yet...)

**Comment by bIgBV** [comment]
* I think including the controls and what the symbols mean in a given page in the popup would be a good start.
* And I love the idea of displaying the meaning behind/expanding on warnings in the future. And it should be fairly straightforward to extend this to perform that function.

File: console/src/view/help.rs
Comment (bIgBV) [inline console/src/view/help.rs]: I wanted to create an area within the popup to render the text into, which would be responsive, but doing it like this doesn't seem to be working.

Comment (hawkw) [reply console/src/view/help.rs]: what isn't working, exactly? is the problem that the text isn't wrapping?

File: console/src/view/help.rs
Comment (hawkw) [inline console/src/view/help.rs]: if you want the text to wrap, you need to call `.wrap()` on the `Paragraph`:
```suggestion
        let display_text = Paragraph::new(content)
            .block(styles.border_block().title("Help"))
            .wrap(Wrap { trim: true });
```

Comment (bIgBV) [reply console/src/view/help.rs]: This was exactly what I was looking for. Thanks!

**Comment by bIgBV** [comment]
@hawkw got the text in the help popup to wrap properly (thanks again for the suggestion). 

image

I also updated the text in the detail views to be (slightly) less generic. I think this is in a good place to be merged and I could separately open a new PR once we finalize the content of the help text for different views.

**Comment by hawkw** [comment]
@bIgBV sorry it took me so long to look at this, I'd really like to actually give it a review now. It looks like this branch is now out of date with `main`, can I get you to rebase or merge `main`?

**Comment by bIgBV** [comment]
@hawkw no worries. I just rebased this branch on top of main.

File: tokio-console/src/view/help.rs
Comment (hawkw) [inline tokio-console/src/view/help.rs:36]: it's kind of weird to me that `help_text` is an `Option` that we have to `take` here? Is there a reason the `render` method has to take an `&mut self`? It could just take `self` --- this isn't a trait method or anything.

File: tokio-console/src/view/help.rs
Comment (hawkw) [inline tokio-console/src/view/help.rs:29]: is there a reason this has these arguments, when it doesn't use them? since this isn't a trait method, I think we should just not take these arguments just because they are passed to other views' `render` methods.
```suggestion
```

File: tokio-console/src/view/mod.rs
Comment (hawkw) [inline tokio-console/src/view/mod.rs:180]: typo: should probably be
```suggestion
    fn handle_help_popup(&mut self, event: crossterm::event::Event) {
```

File: tokio-console/src/view/mod.rs
Comment (hawkw) [inline tokio-console/src/view/mod.rs:185]: nit, take it or leave it: if we used `if ... else if`, we could avoid the second call to `input::is_help_toggle`:
```suggestion
        } else if self.show_help_toggle {
```

File: tokio-console/src/view/mod.rs
Comment (hawkw) [inline tokio-console/src/view/mod.rs:182]: we should probably do this before merging?

File: tokio-console/src/view/mod.rs
Comment (hawkw) [inline tokio-console/src/view/mod.rs:221]: I notice that this will construct the help content for each view *every* time we render a screen, even if we're not showing the help window. That seems a little bit inefficient, since we have to allocate a big `vec` and a bunch of strings.

I think it might be more efficient to add a second `match self.state` inside of the `if self.show_help_toggle` box, and only build the help text if we are actually going to display a help box. That might make the method a bit longer, but I think it's worth doing to avoid having to format the help text when we're not going to display it...

File: tokio-console/src/view/task.rs
Comment (hawkw) [inline tokio-console/src/view/task.rs:272]: should this also include controls?

**Review by hawkw — CHANGES_REQUESTED** [review]
Sorry it took me such a long time to review this, but it looks like a good start. I had some small suggestions.

Thanks for working on this! If you no longer have the time to continue working on it, I could make the remaining suggested changes myself and go ahead and merge this branch. Let me know if you'd like that. Thanks again!

**Comment by hawkw** [comment]
@bIgBV quick ping — still interested in working on this?

**Comment by bIgBV** [comment]
@hawkw I completely missed my notifications on this. While I'm still interested in working on this, my job has kept me pretty busy, and will keep doing so for the next couple of weeks. So if you can pull in my changes and add the ones you suggested on top of that, please go for it!

**Comment by erwanor** [comment]
@bIgBV Hey, would you feel comfortable giving me write access to your fork repo ("Adding contributors" in settings). Otherwise, I'd need to start a fork from scratch and apply your changes etc. which might get a bit gnarly, and we'd lose this convo history.

**Comment by bIgBV** [comment]
@erwanor I've sent you an invitation.

**Comment by hds** [comment]
As no further progress had been made on this PR, a new one was opened (#432) inspired by it and has now been merged.

**Pull request closed without merging**
